lightning/src/chain/mod.rs

```diff
@@ -57,6 +57,81 @@ pub trait Listen {
 	fn block_disconnected(&self, header: &BlockHeader, height: u32);
 }
 
+/// The `Confirm` trait is used to notify when transactions have been confirmed on chain or
+/// unconfirmed during a chain reorganization.
+///
+/// Clients should use this interface instead of [`Listen`] if they receive transaction data only
+/// relevant to that identified by [`Filter`]. For instance, an Electrum client may implement
+/// [`Filter`] by subscribing to activity related to registered transactions and outputs. Upon
+/// notification, it would pass along the relevant transactions using this interface.
+///
+/// The intended use is as follows:
+/// - Call [`transactions_confirmed`] for any on-chain activity of interest.
+/// - Return such transactions from [`get_relevant_txids`] until they have enough confirmations to
+///   no longer be in danger of chain reorganization.
+/// - Call [`transaction_unconfirmed`] for any transaction returned by [`get_relevant_txids`] that
+///   has been reorganized out of the chain.
+/// - Call [`best_block_updated`] whenever a new chain tip becomes available.
+///
+/// See individual method documentation for further details.
```

Comment:
I feel like we need a general "things have to be called in chain order" call-out. I don't think we can or should reasonably handle things out-of-order, and callers of these interface methods need to take special care that they don't call things out of order.